### PR TITLE
Noninteractive run of notebooks and flake8 checks

### PR DESCRIPTION
**Pull request by rijobro**
This uses the python package `papermill` to run notebooks. `papermill` is used instead of `nbconvert --execute` as it displays a progress bar. Without some sort of progress update, it's impossible to tell if execution is slow or is hanging.

Use `sed` to change some variables to 1 to reduce time (e.g., `max_epochs = 180` -> `max_epochs = 1`).

Use jupytext to internally convert from notebook (`*.ipynb`) into script (`*.py`). From here, `flake8` is used for checking PEP8 compliance, and `autopep8`, `isort`, `black` and `autoflake` are used for autofixes.

This PR also contains all the changes to make the notebooks PEP8 compliant.

The most important file to review here is the `runner.sh` script. All other files are hopefully only PEP8 changes, but I would be grateful for any checks to make sure I haven't done anything silly.

Example usage:

```bash

# run for 1 file with autofix
runner.sh --autofix --file modules/load_medical_images.ipynb

# run on all files, skip notebook execution
runner.sh --no-run

```

### CODE REVIEW

**Comment by rijobro** [comment]
fixes https://github.com/Project-MONAI/tutorials/issues/102 
fixes https://github.com/Project-MONAI/tutorials/issues/101
fixes https://github.com/Project-MONAI/tutorials/issues/6
also only uses pip install if the package is missing. Fixes https://github.com/Project-MONAI/tutorials/issues/50

**Comment by rijobro** [comment]
I'm not sure if there is anything valuable in the `.py` files or `runexamples.sh`, beyond just being copies of their corresponding `.ipynb` . If not, we could now delete them.

**Comment by rijobro** [comment]
Fixes https://github.com/Project-MONAI/tutorials/issues/106.

**Comment by wyli** [comment]
could you or someone from the engineering team please help review and validate this PR? @Nic-Ma @yanchengnv @anfeng it is an important one but open for a week without any review.

File: 2d_classification/mednist_tutorial.ipynb [hunk 1]
```diff
@@ -138,8 +125,8 @@
    "source": [
     "## Setup data directory\n",
     "\n",
-    "You can specify a directory with the `MONAI_DATA_DIRECTORY` environment variable.  \n",
```
Comment (Nic-Ma) [inline 2d_classification/mednist_tutorial.ipynb:141]: All these blanks in Markdown doc are to make the next sentence in a new line, that's Markdown gramma I think.
Thanks.

File: acceleration/transform_speed.ipynb [hunk 2]
```diff
@@ -146,7 +138,7 @@
      "name": "stdout",
      "output_type": "stream",
      "text": [
-      "/workspace/data/medical\n"
+      "/home/rbrown/data/MONAI\n"
```
Comment (Nic-Ma) [inline acceleration/transform_speed.ipynb:141]: Please don't use your own path. I didn't check all the path in this PR, could you please help double confirm whether you also changed the path in other places?
Thanks.

File: modules/nifti_read_example.ipynb [hunk 3]
```diff
@@ -138,7 +130,7 @@
      "name": "stdout",
      "output_type": "stream",
      "text": [
-      "/workspace/data/medical\n"
+      "/home/rbrown/data/MONAI\n"
```
Comment (Nic-Ma) [inline modules/nifti_read_example.ipynb:133]: Same issue for path.

**Review by Nic-Ma — COMMENTED** [review]
Thanks very much for the huge effort in this PR.
Overall looks good to me, I didn't check every single line as it's really too big...
And put some comments inline.

Thanks.

File: acceleration/transform_speed.ipynb (hunk 2 above)
Comment (rijobro) [reply acceleration/transform_speed.ipynb]: solved throughout, thanks

File: 2d_classification/mednist_tutorial.ipynb (hunk 1 above)
Comment (rijobro) [reply 2d_classification/mednist_tutorial.ipynb:141]: This is going to be tricky to solve, but thanks for pointing it out.

Comment (rijobro) [reply 2d_classification/mednist_tutorial.ipynb:141]: I've committed a temporary fix f7deb35 pending answer from here: https://github.com/mwouts/jupytext/issues/723.

Comment (Nic-Ma) [reply 2d_classification/mednist_tutorial.ipynb:141]: OK, could you please update the PR to ignore the blanks?

Comment (rijobro) [reply 2d_classification/mednist_tutorial.ipynb:141]: I've done that, but I'll have to revert all these accidental changes manually.

**Comment by Nic-Ma** [comment]
Hi @rijobro ,

Is this PR OK for review again?

Thanks.

**Comment by rijobro** [comment]
hi @Nic-Ma , yes it is now!

**Review by wyli — APPROVED** [review]
thank you @rijobro for looking into these, this is a major enhancement

**Review by Nic-Ma — APPROVED** [review]
Thanks for your quick update.
Looks good to me.

**Pull request merged**